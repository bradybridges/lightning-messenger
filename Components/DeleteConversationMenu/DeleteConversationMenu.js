import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import * as Constants from '../../Constants/Constants';

const DeleteConversationMenu = ({ deleteConversation, toggleDeleteConversationMenu }) => (
  <View style={styles.deleteConversationContainer}>
    <Text style={styles.header}>Delete Conversation?</Text>
    <View style={styles.buttonContainer}>
      <TouchableOpacity style={styles.confirmButton} onPress={deleteConversation}>
        <Text style={styles.textConfirm}>
          Yes
        </Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.cancelButton} onPress={toggleDeleteConversationMenu}>
        <Text style={styles.textCancel}>
          No
        </Text>
      </TouchableOpacity>
    </View>
  </View>
);

const styles = StyleSheet.create({
  deleteConversationContainer: {
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    backgroundColor: Constants.primaryHeaderColor,
    top: '50%',
    left: '50%',
    width: 300,
    height: 150,
    marginTop: -150,
    marginLeft: -150,
    borderRadius: 6,
    zIndex: 5,
    borderColor: Constants.tertiaryBgColor,
    borderWidth: 1,
  },
  buttonContainer: {
    width: '90%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  header: {
    color: Constants.tertiaryBgColor,
    fontSize: 20,
  },
  textConfirm: {
    fontSize: 16,
    color: Constants.primaryHeaderColor,
    fontWeight: 'bold',
  },
  textCancel: {
    fontSize: 20,
    color: Constants.tertiaryBgColor,
    fontWeight: 'bold',
  },
  confirmButton: {
    width: '40%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 35,
    borderRadius: 6,
    backgroundColor: Constants.tertiaryBgColor,
  },
  cancelButton: {
    width: '40%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 35,
    borderRadius: 6,
    borderColor: Constants.tertiaryBgColor,
    borderWidth: 1,
  },
});

export default DeleteConversationMenu;
