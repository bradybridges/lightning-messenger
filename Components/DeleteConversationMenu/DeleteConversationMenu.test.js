import React from 'react';
import { shallow } from 'enzyme';
import DeleteConversationMenu from './DeleteConversationMenu';

describe('deleteConversationMenu', () => {
  const mockDeleteConversation = jest.fn();
  const mockToggleDeleteConversationMenu = jest.fn();
  const wrapper = shallow(<DeleteConversationMenu
    deleteConversation={mockDeleteConversation}
    toggleDeleteConversationMenu={mockToggleDeleteConversationMenu}
  />);

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('pressing the yes button should call deleteConversation', () => {
    wrapper.find('TouchableOpacity').at(0).simulate('press');
    expect(mockDeleteConversation).toHaveBeenCalled();
  });

  it('pressing the no button should call toggleDeleteConversationMenu', () => {
    wrapper.find('TouchableOpacity').at(1).simulate('press');
    expect(mockToggleDeleteConversationMenu).toHaveBeenCalled();
  });
});
