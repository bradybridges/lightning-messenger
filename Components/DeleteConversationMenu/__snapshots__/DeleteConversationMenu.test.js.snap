// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteConversationMenu should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#121716",
      "borderColor": "#6E8F76",
      "borderRadius": 6,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": 150,
      "justifyContent": "space-evenly",
      "left": "50%",
      "marginLeft": -150,
      "marginTop": -200,
      "position": "absolute",
      "top": "50%",
      "width": 300,
      "zIndex": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#6E8F76",
        "fontSize": 24,
      }
    }
  >
    Delete Conversation?
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "width": "90%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6E8F76",
          "borderRadius": 6,
          "display": "flex",
          "height": 35,
          "justifyContent": "center",
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#121716",
            "fontSize": 20,
          }
        }
      >
        Yes
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6E8F76",
          "borderRadius": 6,
          "borderWidth": 1,
          "display": "flex",
          "height": 35,
          "justifyContent": "center",
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6E8F76",
            "fontSize": 20,
          }
        }
      >
        No
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
